tests/test_dependencies.py

```diff
@@ -69,20 +75,43 @@ def test_has_at_most_one_pinned_dependency(
 
         filtered_requirements.append(req)
 
-    assert (
-        len(filtered_requirements) <= 1
-    ), f"Expected no more than one pin.\n{pprint.pformat(environment)}"
-
-
-def test_valid_numpy_is_installed():
-    filtered_requirements = []
-    for req in get_package_dependencies():
-        if req.marker.evaluate():
-            filtered_requirements.append(req)
-
-    assert (len(filtered_requirements) == 1), "Expected exactly one pin."
-
-    item, = filtered_requirements[0].specifier
-
-    import numpy
-    assert item.version == numpy.__version__
+    # since we cannot run the installation tests on all platforms,
+    # we formulate the conditions when we expect a pin
+    expect_pin = False
+    match (platform_system, platform_machine):
+        case [("Linux" | "Darwin" | "Windows"), "x86_64"]:
+            expect_pin = True  # baseline
+        case [("Linux" | "Windows"), "x86"]:
+            # 32 bit wheels on Linux only until numpy 1.21, but should still be compatible
+            expect_pin = True
+        case ["Linux", "aarch64"]:
+            expect_pin = True  # as of 1.19.2
+        case ["Darwin", "arm64"]:
+            expect_pin = True  # as of 1.21
+        # no official wheels, but register minimum compatibility
+        case ["Linux", "ppc64le"]:
+            expect_pin = True  # support +/- equivalent to Linux/aarch64, but no wheels
+        case ["Linux", "s390x"]:
+            expect_pin = True  # as of 1.17.5
+        case ["Linux", "loongarch64"]:
+            expect_pin = True  # as of 1.22
```

Comment:
I find this a very compact and readable crystallization of what we're trying to achieve in the setup.cfg, in addition to the distinction of whether wheels are available or not (we could add more testing for this if we whip up some emulation, see #73).